packages/dds/tree/src/domains/json/typedJsonCursor.ts

```diff
@@ -0,0 +1,94 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Jsonable } from "@fluidframework/datastore-definitions";
+import { SynchronousNavigationResult, TreeNavigationResult } from "../../forest";
+import { lookupGlobalFieldSchema, lookupTreeSchema, NamedTreeSchema, StoredSchemaRepository, ValueSchema } from "../../schema-stored";
+import { FieldKey, keyFromSymbol, TreeType } from "../../tree";
+import { brand } from "../../util";
+import { isPrimitive } from "../../feature-libraries";
+import { JsonCursor } from "./jsonCursor";
+
+/**
+ * An extension of JsonCursor, which allows to use simple schemas having no polymorphic nor sequence fields.
+ * Used in {@link EditableTree} to support json over JsonableTree as the input type.
+ */
+// TODO: add support for a custom type resolution.
+export class TypedJsonCursor<T> extends JsonCursor<T> {
```

Comment:
As this is a new cursor implementation, it should implement the new cursor API, instead of the old one we are in the process of migrating away from. For the short term, and code that needs the old API to be implemented for it can do so by converting it through a format that supposers both (ex: MapTree).

Also I'd recommend basing it off of treeTextCursor.ts (which is the new API, and does not use jsonable cursor). Having a clear, documented, input type (in that case JsonableTree) is important and missing from this one. Currently the doc comment on this type seems to be the only documentation of the input data format, and thats not enough.

Also it should be hooked into the cursor test suites. As far as I can tell this has no tests.

Thus cursor should located elsewhere: since its not specific to any domain.

it should probably be in its own PR to ease review.